userspace/libscap/engine/savefile/scap_savefile.c

```diff
@@ -1605,14 +1605,47 @@ static int32_t scap_read_fdlist(scap_reader_t *r,
 	return SCAP_SUCCESS;
 }
 
-static int32_t scap_read_section_header(scap_reader_t *r, char *error) {
+#define MIN_BLOCK_SIZE ((sizeof(block_header) + sizeof(uint32_t)))
+#define MIN_SECTION_HEADER_SIZE (MIN_BLOCK_SIZE + sizeof(section_header_block))
+
+static int32_t scap_read_section_header(scap_reader_t *r,
+                                        uint32_t block_total_length,
+                                        char *error) {
 	section_header_block sh;
 	uint32_t bt;
 
 	//
 	// Read the section header block
 	//
-	if(r->read(r, &sh, sizeof(sh)) != sizeof(sh) || r->read(r, &bt, sizeof(bt)) != sizeof(bt)) {
+	if(r->read(r, &sh, sizeof(sh)) != sizeof(sh)) {
+		snprintf(error, SCAP_LASTERR_SIZE, "error reading from file (1)");
```

Comment:
Do you mind to use `scap_errprintf` for error handling? Just like in https://github.com/falcosecurity/libs/pull/2286 . It's the same for the other 2 occurrences that you are adding.

Reply:
Done.